tests/e2e/cairo1/test_cairo1_testing.py

```diff
@@ -16,3 +18,28 @@ def test_no_tests_found(protostar: ProtostarFixture):
     result = protostar(["test-cairo1", "tests"])
 
     assert "No test cases found" in result
+
+
+def test_failing_tests(protostar: ProtostarFixture, copy_fixture: CopyFixture):
+    copy_fixture("cairo1_project", "./cairo1_project")
+    template_test_files = Path("cairo1_project/tests").glob("**/*")
+    for template_test_file in template_test_files:
+        template_test_file.unlink()
+    copy_fixture("cairo1/failing_test.cairo", "./cairo1_project/tests")
+    os.chdir("./cairo1_project")
+
+    result = protostar(["--no-color", "test-cairo1", "tests"], ignore_exit_code=True)
+
+    expected_output_lines = [
+        "Collected 1 suite, and 3 test cases",
+        "[PASS] tests/failing_test.cairo test_ok",
+        "[FAIL] tests/failing_test.cairo test_panic_multiple_values",
+        "Test failed with data: [101, 102, 103]",
+        "[FAIL] tests/failing_test.cairo test_panic_single_value",
+        "Test failed with data: [21]",
+        "1 failed, 1 total",
+        "2 failed, 1 passed, 3 total",
+    ]
+
+    for expected_output_line in expected_output_lines:
```

Comment:
[nitpick] I actually don't think we need to assert the whole content, just a few bits would be okay (is the failure data included, are the functions collected, how many have failed and passed)

This is fine as-is though

Reply:
Hmm, I think I covered pretty much what you mentioned + what failed and what passed, I think it's okay, it's not a big overhead or something.